geti_sdk/rest_clients/credit_system_client.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import logging
+from typing import List, Optional, Union
+
+from geti_sdk.data_models import Subscription
+from geti_sdk.data_models.job import Job, JobCost
+from geti_sdk.http_session import GetiSession
+from geti_sdk.utils.job_helpers import get_job_by_id
+from geti_sdk.utils.serialization_helpers import deserialize_dictionary
+from geti_sdk.utils.workspace_helpers import get_default_workspace_id
+
+
+def allow_supported(func):
+    """
+    Decorate the class methods to allow them to run only if the Credit System is supported.
+    """
+
+    def wrapper(instance, *args, **kwargs):
+        if instance._is_supported:
+            return func(instance, *args, **kwargs)
+        else:
+            logging.warning(
+                "Credit System is not supported by the Intel Geti Platform."
+            )
+            return None
+
+    return wrapper
+
+
+class CreditSystemClient:
+    """
+    Class to work with credits in Intel Geti.
+    """
+
```

Comment:
It would be nice to also have a method to get credit accounts information (example below)
```json
{
  "total_matched": 2,
  "next_page": null,
  "credit_accounts": [
    {
      "id": "b9477b6f-1b65-4fd4-a52d-f58e9547e58a",
      "organization_id": "95fa9db8-f3c6-4cc7-aa83-49e695adb326",
      "name": "Freemium Credits",
      "renewable_amount": 1000,
      "renewal_day_of_month": 29,
      "created": 1722253713258,
      "updated": 1722253713258,
      "expires": null,
      "balance": {
        "incoming": 0,
        "available": 0,
        "blocked": 0
      }
    },
    {
      "id": "b84d8471-846b-40b8-8bd7-5c3450a88c9e",
      "organization_id": "95fa9db8-f3c6-4cc7-aa83-49e695adb326",
      "name": "Welcoming Credits",
      "renewable_amount": 0,
      "renewal_day_of_month": null,
      "created": 1722253713235,
      "updated": 1722253713235,
      "expires": 1724932113234,
      "balance": {
        "incoming": 3000,
        "available": 2660,
        "blocked": 70
      }
    }
  ]
}
```

Reply:
Added a method for that!